ComfortZoneApp/app/src/main/java/com/example/comfortzone/ui/HostActivity.java

```diff
@@ -96,37 +109,49 @@ private void createFragments() {
     private void listenerSetup() {
         bottomNavigationView.setOnItemSelectedListener(item -> {
             Pair<Integer, Integer> animations;
-            switch (item.getItemId()) {
-                case R.id.action_flight:
-                    fragment = flightFragment;
-                    animations = setAnimationLeftToRight();
-                    break;
-                case R.id.action_input:
-                    if (fragment == flightFragment) {
-                        animations = setAnimationRightToLeft();
-                    } else {
-                        animations = setAnimationLeftToRight();
-                    }
-                    fragment = inputFragment;
-                    break;
-                case R.id.action_profile:
-                default:
-                    fragment = profileFragment;
-                    animations = setAnimationRightToLeft();
-                    break;
+            if (isLoading) {
+                animations = setAnimationLeftToRight();
+                fragment = new LoadingFragment();
+            } else {
+                animations = bottomNavSelected(item);
             }
             fragmentManager.beginTransaction()
                     .setCustomAnimations(
                             animations.first,
                             animations.second
                     )
-                    .replace(R.id.flContainer, fragment, "input")
+                    .replace(R.id.flContainer, fragment)
                     .commit();
             return true;
         });
         bottomNavigationView.setSelectedItemId(R.id.action_profile);
     }
 
+    @NonNull
+    private Pair<Integer, Integer> bottomNavSelected(MenuItem item) {
+        Pair<Integer, Integer> animations;
+        switch (item.getItemId()) {
+            case R.id.action_flight:
+                fragment = flightFragment;
+                animations = setAnimationLeftToRight();
+                break;
+            case R.id.action_input:
+                if (fragment == flightFragment) {
```

Comment:
Another way to this if else is using a ternary operator just as an FYI. More details about the ternary operator are here https://www.baeldung.com/java-ternary-operator

animations = fragment == flightFragment ? setAnimationRightToLeft() : setAnimationLeftToRight()